ethernet: stm32: add a alternative HAL_ETH_Init for V1 api

maass-hamburg opened the pull request:
~~add a alternative HAL_ETH_Init for V1 api without phy
configuration, as this is already been done outside of
the ethernet driver.~~

Deactivates the phy config and init in the hal code, because the phy driver in zephyr is already responsible for it.

erwango commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -877,6 +877,75 @@ static void RISAF_Config(void)
+#if defined(CONFIG_ETH_STM32_HAL_API_V1)
+#define ETH_TIMEOUT_SWRESET 500U
+
+HAL_StatusTypeDef HAL_ETH_Init_alt(ETH_HandleTypeDef *heth)
```
It is fine not using `HAL_ETH_Init`, but please don't reimplement a HAL function and instead implement as much as possible a zephyr function:
Don't call it HAL_... but use zephyr function name
Don't return HAL type and take dev as argument.
Access ETH registers instead of `(heth->Instance)->`
....
Likely the 2 only pure HAL things that should remain are the `State` and `Lock` updates for further HAL operations.

clamattia reviewed (commented):
Side note. Why is `eth_stm32_init_v1_api` happening at driver init. And the equivalent for `v2` at `api->init` (called by net core)?

maass-hamburg commented:
> Side note. Why is `eth_stm32_init_v1_api` happening at driver init. And the equivalent for `v2` at `api->init` (called by net core)?

v2 got moved there in https://github.com/zephyrproject-rtos/zephyr/pull/86621. the phy specific code from that than got removed in my https://github.com/zephyrproject-rtos/zephyr/pull/87593. Maybe it can be moved back? @marwaiehm-st what do you think, it was your pr?

clamattia commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -990,28 +993,70 @@ static int eth_initialize(const struct device *dev)
 
-#if defined(CONFIG_ETH_STM32_HAL_API_V1)
+#if defined(CONFIG_ETH_STM32_HAL_API_V2)
+	HAL_StatusTypeDef hal_ret = HAL_OK;
```
Maybe we could have an equivalent function like `eth_stm32_init_v1_api` here call it `eth_stm32_init_v2_api`?

maass-hamburg replied:
eth_stm32_init_v1_api() is a replacement for HAL_ETH_Init(), because HAL_ETH_Init() in the v1 api contains code that will configure and access the phy, that we don't want, it's mainly the code from the hal adapted to zephyr and the unwanted parts are removed. and for maintainability I'd like keep it like it.

clamattia replied:
Ok

clamattia reviewed (approved):
Thank you for moving the `v1` and `v2` functions closer together. I have added another small hint. I am not familiar enough with the hal to review the `eth_stm32_init_v1_api`. From what I can see looks good.

maass-hamburg commented:
ping @erwango

etienne-lms reviewed (commented):
Some minor comments

maass-hamburg commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -845,6 +842,65 @@ static void RISAF_Config(void)
+	/* Get the ETHERNET MACMIIAR value, this is responsible for mdio, so save it to restore it
+	 * later
+	 */
+	tmpreg1 = (heth->Instance)->MACMIIAR;
```
done

maass-hamburg commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -845,6 +842,65 @@ static void RISAF_Config(void)
+
+	/* Wait for software reset */
+	if (WAIT_FOR((((heth->Instance)->DMABMR & ETH_DMABMR_SR) == (uint32_t)RESET),
+		     ETH_TIMEOUT_SWRESET, NULL) == false) {
```
done

maass-hamburg commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -845,6 +842,65 @@ static void RISAF_Config(void)
+		return -ETIMEDOUT;
+	}
+
+	(heth->Instance)->MACMIIAR = (uint32_t)tmpreg1;
```
done

maass-hamburg commented:
@erwango ping

maass-hamburg committed:
ethernet: stm32: remove phy init and config from HAL_ETH_Init

the phy is already configured by the phy driver in zephyr
and we don't want to overwrite it in the HAL_ETH_Init
function in the stm32 HAL.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

maass-hamburg commented:
@erwango PTAL

maass-hamburg commented:
rebased and updated west.yml

etienne-lms reviewed (approved):
LGTM
